### PR TITLE
Trb 31

### PR DESCRIPTION
**Pull request by ChrisFrill**
Probably have to refactor and connect with ProgressionService. Especially the purchase and upgrade of troops, if the user can buy 1 or more troop by one request.

### CODE REVIEW

**Review by lyancsie — APPROVED** [review]
Refactor may be needed, but it seems to be a good start

File: src/main/java/com/greenfox/tribes1/Kingdom/Kingdom.java [hunk 1]
```diff
@@ -4,16 +4,14 @@
 import com.greenfox.tribes1.Building.Building;
 import com.greenfox.tribes1.Resources.KingdomResource;
 import com.greenfox.tribes1.Troop.Model.Troop;
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Getter;
-import lombok.Setter;
+import lombok.*;
```
Comment (pitner33) [inline src/main/java/com/greenfox/tribes1/Kingdom/Kingdom.java]: wasn't the case that we don't use lombok.* but the separate lombok.getter...etc ones?

File: src/main/java/com/greenfox/tribes1/Purchase/PurchaseService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    this.troopService = troopService;
+  }
+
+  public void purchase(Kingdom kingdom, Long id, Long upgradeCost) throws GoldNotEnoughException, NotValidResourceException {
+    List<KingdomResource> kingdomResources = kingdom.getResources();
+    Gold gold = getGoldAmount(kingdomResources);
+    purchaseIfPossible(gold, 1L, upgradeCost);
+  }
+
+  public void upgradeTroop(Kingdom kingdom, Long id) throws GoldNotEnoughException, NotValidResourceException, TroopIdNotFoundException {
```
Comment (pitner33) [inline src/main/java/com/greenfox/tribes1/Purchase/PurchaseService.java]: rename pls

File: src/main/java/com/greenfox/tribes1/Purchase/PurchaseService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    Troop troop = troopService.findById(id);
+    List<KingdomResource> kingdomResources = kingdom.getResources();
+    Gold gold = getGoldAmount(kingdomResources);
+    Long upgradeTo = troop.getLevel() + 1L;
+    Long troopUpgradeCost = 10L;
+
+    purchaseIfPossible(gold, upgradeTo, troopUpgradeCost);
+  }
+
+  public void upgradeBuilding(Kingdom kingdom, Long id) throws GoldNotEnoughException, BuildingIdNotFoundException, NotValidResourceException {
```
Comment (pitner33) [inline src/main/java/com/greenfox/tribes1/Purchase/PurchaseService.java]: rename pls

**Review by csadesz — APPROVED** [review]

**Pull request merged**